src/main/java/database/service/service/JsonDatabaseService.java

```diff
@@ -0,0 +1,195 @@
+package database.service.service;
+
+import code.practice.exceptions.CriticalDatabaseException;
+import code.practice.exceptions.DatabaseDoesNotExistException;
+import code.practice.exceptions.ProvidedIdDoesNotExistException;
+import com.fasterxml.jackson.databind.JavaType;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import database.service.entity.BaseEntity;
+import database.service.helper.Settings;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class JsonDatabaseService implements DatabaseService {
+    private final Settings settings = new Settings();
+    private final ObjectMapper objectMapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    private Map<String, Integer> entityIds = new HashMap<>();
+
+    public JsonDatabaseService() throws CriticalDatabaseException {
+    }
+
+    @Override
+    public boolean createTable(Class<? extends BaseEntity> entityClass) throws IOException {
+        File jsonDatabaseFile = new File(getDatabasePath(entityClass));
+        jsonDatabaseFile.createNewFile();
+        Files.writeString(jsonDatabaseFile.toPath(), "[]");
```

Comment:
Please write to file only if it's successfuly created, otherwise throw exception.